autotest/ogr/ogr_mdb.py

```diff
@@ -0,0 +1,216 @@
+#!/usr/bin/env pytest
+# -*- coding: utf-8 -*-
+###############################################################################
+# $Id$
+#
+# Project:  GDAL/OGR Test Suite
+# Purpose:  Test read functionality for OGR MDB driver.
+# Author:   Even Rouault <even dot rouault at spatialys.com>
+#
+###############################################################################
+# Copyright (c) 2010-2012, Even Rouault <even dot rouault at spatialys.com>
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+###############################################################################
+
+import os
+from osgeo import ogr
+
+
+import gdaltest
+import ogrtest
+import pytest
+
+
+@pytest.fixture(scope="module", autouse=True)
+def setup_driver():
+    driver = ogr.GetDriverByName('MDB')
+    if driver is not None:
+        driver.Register()
+    else:
+        pytest.skip("MDB driver not available", allow_module_level=True)
+
+    # remove PGeo driver
+    ogrtest.pgeo_driver = ogr.GetDriverByName('PGeo')
+    if ogrtest.pgeo_driver is not None:
+        ogrtest.pgeo_driver.Deregister()
+
```

Comment:
the cleanest way would be to add a ``yield`` just at the end of setup_driver() and followed by the content of the test_ogr_mdb_cleanup() (reregistration of the pgeo_driver, which you would need to attach to the ogrtest). See ogr_kml.py for a similar example
The advantage of this is that it could allow those tests to run with the --random-order switch of pytest

Reply:
I'm confused why the pgeo driver would need to be attached to ogrtest in that setup -- couldn't it remain as a local variable only?

Reply:
oops, sorry I forgot the negation in my sentence and actually meant that it can remain a local variable